clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1398,6 +1409,100 @@ event_delay_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t
     return DR_EMIT_DEFAULT;
 }
 
+
+typedef struct _func_metadata {
+    const trace_marker_type_t marker_retaddr;
+    const trace_marker_type_t marker_arg;
+    const trace_marker_type_t marker_retval;
+    const char *name;
+    const int num_args;
+} func_metadata;
+
+#define MALLOC_RETADDR TRACE_MARKER_TYPE_FUNC_MALLOC_RETADDR
+#define MALLOC_ARG     TRACE_MARKER_TYPE_FUNC_MALLOC_ARG
+#define MALLOC_RETVAL  TRACE_MARKER_TYPE_FUNC_MALLOC_RETVAL
+
+static const func_metadata instru_funcs[] = {
+    {MALLOC_RETADDR, MALLOC_ARG, MALLOC_RETVAL, "malloc", 1},
+    {MALLOC_RETADDR, MALLOC_ARG, MALLOC_RETVAL, "(anonymous namespace)::do_malloc", 1},
+
+#ifdef WINDOWS
+    {MALLOC_RETADDR, MALLOC_ARG, MALLOC_RETVAL, "malloc_impl", 1 },
+#endif // WINDOWS
+
+#ifdef UNIX
+    {MALLOC_RETADDR, MALLOC_ARG, MALLOC_RETVAL, "tc_malloc", 1 },
+    {MALLOC_RETADDR, MALLOC_ARG, MALLOC_RETVAL, "__libc_malloc", 1},
```

Comment:
Intercepting malloc & co. is a complex topic.  Xref Dr. Memory's interception of all heap allocation routines (https://github.com/DynamoRIO/drmemory/blob/master/common/alloc.c#L317,
https://github.com/DynamoRIO/drmemory/blob/master/common/alloc.c#L433) and longstanding plans to turn that into a library (https://github.com/DynamoRIO/drmemory/issues/824) though that has yet to happen.

Please include comments explaining that this is a first pass and is not complete.

Reply:
Done